types/src/account_state/mod.rs

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::account_config::{AccountResource, ACCOUNT_RESOURCE_PATH};
+use anyhow::{Error, Result};
+use serde::export::Formatter;
+use serde::{Deserialize, Serialize};
+use std::collections::btree_map::BTreeMap;
+use std::convert::TryFrom;
+use std::fmt;
+use std::ops::{Deref, DerefMut};
+
+#[derive(Default, Deserialize, Serialize)]
+pub struct AccountState(BTreeMap<Vec<u8>, Vec<u8>>);
+
+impl AccountState {
+    pub fn get_account_resource(&self) -> Result<Option<AccountResource>> {
+        self.0
+            .get(&*ACCOUNT_RESOURCE_PATH)
+            .map(|bytes| lcs::from_bytes(bytes))
+            .transpose()
+            .map_err(Into::into)
+    }
+}
+
+impl Deref for AccountState {
+    type Target = BTreeMap<Vec<u8>, Vec<u8>>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl DerefMut for AccountState {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.0
+    }
+}
```

Comment:
I'd recommend not implementing these and add a few methods to the `impl` block.

Reply:
k.. I was lazy.